### PR TITLE
🏃clusterctl: more cleanups

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
Some cleanups:
- Drops `NewOptions` structs and use internal types instead
- Drops `Cluster.ProviderObjects` struct (not used)

**Which issue(s) this PR fixes**:
/area clusterctl
/assign @ncdc
/assign @vincepri

### CODE REVIEW

File: cmd/clusterctl/pkg/client/client.go [hunk 1]
```diff
@@ -151,36 +151,29 @@ type ClusterClientFactory func(string) (cluster.Client, error)
...
-	injectConfig            config.Client
-	injectRepositoryFactory RepositoryClientFactory
-	injectClusterFactory    ClusterClientFactory
-}
-
 // Option is a configuration option supplied to New
-type Option func(*NewOptions)
+type Option func(*clusterctlClient)
 
 // InjectConfig implements a New Option that allows to override the default configuration client used by clusterctl.
```
Comment (wfernandes) [inline cmd/clusterctl/pkg/client/client.go]:
```suggestion
// InjectConfig allows to override the default configuration client used by clusterctl.
```

File: cmd/clusterctl/pkg/client/client.go [hunk 2]
```diff
@@ -151,36 +151,29 @@ type ClusterClientFactory func(string) (cluster.Client, error)
...
 func InjectConfig(config config.Client) Option {
-	return func(c *NewOptions) {
-		c.injectConfig = config
+	return func(c *clusterctlClient) {
+		c.configClient = config
 	}
 }
 
 // InjectRepositoryFactory implements a New Option that allows to override the default factory used for creating
```
Comment (wfernandes) [inline cmd/clusterctl/pkg/client/client.go]:
```suggestion
// InjectRepositoryFactory allows to override the default factory used for creating
```

File: cmd/clusterctl/pkg/client/client.go [hunk 3]
```diff
@@ -151,36 +151,29 @@ type ClusterClientFactory func(string) (cluster.Client, error)
...
 // RepositoryClient objects.
 func InjectRepositoryFactory(factory RepositoryClientFactory) Option {
-	return func(c *NewOptions) {
-		c.injectRepositoryFactory = factory
+	return func(c *clusterctlClient) {
+		c.repositoryClientFactory = factory
 	}
 }
 
 // InjectClusterClientFactory implements a New Option that allows to override the default factory used for creating
```
Comment (wfernandes) [inline cmd/clusterctl/pkg/client/client.go]:
```suggestion
// InjectClusterClientFactory allows to override the default factory used for creating
```

File: cmd/clusterctl/pkg/client/cluster/client.go [hunk 4]
```diff
@@ -120,35 +112,28 @@ func (c *clusterClient) ObjectMover() ObjectMover {
...
-	injectProxy                   Proxy
-	injectRepositoryClientFactory RepositoryClientFactory
-	injectPollImmediateWaiter     PollImmediateWaiter
-}
-
 // Option is a configuration option supplied to New
-type Option func(*NewOptions)
+type Option func(*clusterClient)
 
 // InjectProxy implements a New Option that allows to override the default proxy used by clusterctl.
```
Comment (wfernandes) [inline cmd/clusterctl/pkg/client/cluster/client.go]:
```suggestion
// InjectProxy implements allows to override the default proxy used by clusterctl.
```

File: cmd/clusterctl/pkg/client/cluster/client.go [hunk 5]
```diff
@@ -120,35 +112,28 @@ func (c *clusterClient) ObjectMover() ObjectMover {
...
 func InjectProxy(proxy Proxy) Option {
-	return func(c *NewOptions) {
-		c.injectProxy = proxy
+	return func(c *clusterClient) {
+		c.proxy = proxy
 	}
 }
 
 // InjectRepositoryFactory implements a New Option that allows to override the default factory used for creating
```
Comment (wfernandes) [inline cmd/clusterctl/pkg/client/cluster/client.go]:
```suggestion
// InjectRepositoryFactory allows to override the default factory used for creating
```

File: cmd/clusterctl/pkg/client/cluster/client.go [hunk 6]
```diff
@@ -120,35 +112,28 @@ func (c *clusterClient) ObjectMover() ObjectMover {
...
 // RepositoryClient objects.
 func InjectRepositoryFactory(factory RepositoryClientFactory) Option {
-	return func(c *NewOptions) {
-		c.injectRepositoryClientFactory = factory
+	return func(c *clusterClient) {
+		c.repositoryClientFactory = factory
 	}
 }
 
 // InjectPollImmediateWaiter implements a New Option that allows to override the default PollImmediateWaiter used by clusterctl.
```
Comment (wfernandes) [inline cmd/clusterctl/pkg/client/cluster/client.go]:
```suggestion
// InjectPollImmediateWaiter allows to override the default PollImmediateWaiter used by clusterctl.
```

File: cmd/clusterctl/pkg/client/config/client.go [hunk 7]
```diff
@@ -49,18 +49,13 @@ func (c *configClient) Variables() VariablesClient {
...
-// NewOptions carries the options supported by New
-type NewOptions struct {
-	injectReader Reader
-}
-
 // Option is a configuration option supplied to New
-type Option func(*NewOptions)
+type Option func(*configClient)
 
 // InjectReader implements a New Option that allows to override the default configuration reader used by clusterctl.
```
Comment (wfernandes) [inline cmd/clusterctl/pkg/client/config/client.go]:
```suggestion
// InjectReader allows to override the default configuration reader used by clusterctl.
```

**Comment by fabriziopandini** [comment]
@wfernandes comment addressed

**Pull request merged**